tests/banks-v1.test.js

```diff
@@ -33,11 +33,30 @@ describe('banks v1 (E2E)', () => {
     });
   });
 
-  test('GET /banks/v1', async () => {
-    const requestUrl = `${global.SERVER_URL}/api/banks/v1`;
-    const response = await axios.get(requestUrl);
+  describe('GET /banks/v1/:code', () => {
+    test('Listar todos os bancos', async () => {
+      const requestUrl = `${global.SERVER_URL}/api/banks/v1`;
+      const response = await axios.get(requestUrl);
+
+      expect(response.status).toBe(200);
+      expect(Array.isArray(response.data)).toBe(true);
+    });
+
+    test('Listar bancos por parte de nome ou código', async () => {
+      const queryName = 'inter';
 
-    expect(response.status).toBe(200);
-    expect(Array.isArray(response.data)).toBe(true);
+      const requestUrl = `${global.SERVER_URL}/api/banks/v1?search=${queryName}`;
+      const response = await axios.get(requestUrl);
+
+      const filterNamesFromBanks = response.data.map(
+        ({ name }) =>
+          !!String(name).toLowerCase().includes(queryName.toLowerCase())
+      );
+
+      expect(response.status).toBe(200);
+      expect(
+        filterNamesFromBanks.every((nameCheck) => nameCheck === true)
+      ).toBe(true);
+    });
```

Comment:
Pra falar a verdade esse teste ficou bem confuso pra mim, talvez por conta das nomenclaturas, ou pela forma que foi escrito, o que acha de deixar ele mais limpo e claro? **SUGESTÃO**:

```suggestion
      const requestUrl = `${global.SERVER_URL}/api/banks/v1?search=${queryName}`;
      const response = await axios.get(requestUrl);

      const expectedResponse = [
         {
             code: 'bla',
             name: 'inter bla bla'
         },
         ...
      ]
      expect(response.status).toBe(200);
      expect(response.body).toBe(expectedResponse);
    });
```

Reply:
Este serviço trabalha com os dados atualizados recuperados pelo csv do Bacen. Implementar com assert com dados estáticos, como você sugere, seria bem interesse se a funcionalidade trabalhasse com dados mockados, onde o teste tivesse pleno controle sobre o dado que está sendo inserido, e consequentemente recuperado durante o teste. 

Entretanto, assim como está qualquer mínima alteração que exista nesta base de dados irá refletir na resposta dos testes. o que não seria o teste específico da funcionalidade em si, e sim da convergência das informações com a base do Bacen. 

Um exemplo simples que quebraria esse teste da sua sugestão seria simples mudança do nome do banco de "inter" (todo minúsculo) para "Inter" (com a primeira letra maiúscula). A funcionalidade ainda estaria plenamente operacional, mas iria quebrar o teste por essa variação da base de dados. PS: e convenhamos, a base de dados vem de uma planilha, que não garante muito a padronização deste dado.

Se tratando de uma origem dinâmica, fiz um simples map para percorrer o array, converter o name sempre para minúsculas e comparar com o nome buscado, retornando um array de booleanos (true caso o nome buscado seja parte do nome do banco retornado). O assert verifica se temos um array de "trues", ou seja, se algum nome de banco retornado não conter o nome procurado o teste irá falhar.

Analogamente, como você havia sugerido implementar o teste para a busca parcial de código bancário, na implementação com assert de dados estáticos, caso exista a inclusão de um novo banco com o mesmo trecho, o teste irá falhar (mesmo atendendo plenamente as suas funções). Exemplo prático, hoje se é feita uma query com search=77, irá retornar:

```json
  {
    "ispb": "00416968",
    "name": "BANCO INTER",
    "code": 77,
    "fullName": "Banco Inter S.A."
  },
  {
    "ispb": "17826860",
    "name": "BMS SCD S.A.",
    "code": 377,
    "fullName": "BMS SOCIEDADE DE CRÉDITO DIRETO S.A."
  },
  {
    "ispb": "33042953",
    "name": "CITIBANK N.A.",
    "code": 477,
    "fullName": "Citibank N.A."
  },
  {
    "ispb": "65913436",
    "name": "GUIDE",
    "code": 177,
    "fullName": "Guide Investimentos S.A. Corretora de Valores"
  }
]
```

Se um banco começar a operar com o código 577, ele passaria a entrar nessa lista e falharia o teste.

Implementei os testes para que eles resistam a alteração da padronização dos dados. Acredito que desta maneira ajudaria muito a não se gastar horas no futuro buscando por erros na função que não existem (e sim é fruto da mudança dos dados).

De qualquer maneira, implementar com dados estáticos é absurdamente rápido e fácil. Se é a maneira como preferem, irei atualizar.